cmd/disk-cache-utils.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/minio/minio/cmd/crypto"
+)
+
+type cacheControl struct {
+	exclude  bool
+	expiry   time.Time
+	maxAge   int
+	sMaxAge  int
+	minFresh int
+}
+
+// cache exclude directives in cache-control header
+var cacheExcludeDirectives = []string{
+	"no-cache",
+	"no-store",
+	"must-revalidate",
+}
+
+// returns true if cache exclude directives are set.
+func isCacheExcludeDirective(s string) bool {
+	for _, directive := range cacheExcludeDirectives {
+		if s == directive {
+			return true
+		}
+	}
+	return false
+}
+
+// returns struct with cache-control settings from user metadata.
+func getCacheControlOpts(m map[string]string) (c cacheControl, err error) {
```

Comment:
May be obvious to some users, but can we add a section in document about the supported cache control headers?

Reply:
updated docs for this - @nitisht, PR needs to be reapproved.